### PR TITLE
Log total time that the query spent in the queue

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

Collect and log total time that query (and all subqueries) spent in the queue inside query-frontend or query-scheduler.

(First commit contains all necessary changes for feature to work, second commit has unit tests updates.)

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/frontend/v1/frontend_test.go
Comment (dimitarvdimitrov) [inline pkg/frontend/v1/frontend_test.go:231]: there's very little to test in pkg/frontend/v1/frontend.go. Isn't this test testing more or less the same thing as `TestHandler_ServeHTTP`? I added some tests in https://github.com/grafana/mimir/pull/6473 which make it possible to assert that the actual logged value - see `TestHandler_LogsFormattedQueryDetails`. Do you think that can be reused?

File: pkg/querier/stats/stats_test.go
Comment (dimitarvdimitrov) [inline pkg/querier/stats/stats_test.go:127]: wall -> queue

File: pkg/querier/worker/frontend_processor_test.go
Comment (dimitarvdimitrov) [inline pkg/querier/worker/frontend_processor_test.go:140]: query time or queue time?

**Review by dimitarvdimitrov — APPROVED** [review]
that's great! thanks for adding it! I worked on a similar thing in another PR and we might be able to reuse the tests (once that's merged)

File: pkg/frontend/v1/frontend_test.go
Comment (pstibrany) [reply pkg/frontend/v1/frontend_test.go:232]: This test is slightly different from `TestHandler_ServeHTTP` in that handler modifies value in the stats, and then we check if it was reported correctly. I admit that it would probably be better if it was moved to `handler_test.go` and tested other reported values from stats too. If you agree, I would do that.

`TestHandler_LogsFormattedQueryDetails` indeed seems to be similar, although it's testing `QueryDetails`, not stats.

File: pkg/querier/stats/stats_test.go
Comment (pstibrany) [reply pkg/querier/stats/stats_test.go]: Fixed in https://github.com/grafana/mimir/pull/6537/commits/0c1786c77965bfd62692ae09efce2beb13f54422.

File: pkg/querier/worker/frontend_processor_test.go
Comment (pstibrany) [reply pkg/querier/worker/frontend_processor_test.go:140]: Fixed in https://github.com/grafana/mimir/pull/6537/commits/0c1786c77965bfd62692ae09efce2beb13f54422.

File: pkg/frontend/v1/frontend_test.go
Comment (dimitarvdimitrov) [reply pkg/frontend/v1/frontend_test.go:232]: a QueryDetails also contains QueryStats for this purpose. I tend to prefer more complicated and elaborate unit tests instead of individual ones. I'll leave this up to

Comment (pstibrany) [reply pkg/frontend/v1/frontend_test.go:232]: I'll merge this as-is for now, and once https://github.com/grafana/mimir/pull/6473 is in, I will try to take a closer look at reusing `TestHandler_LogsFormattedQueryDetails`.

**Pull request merged**
